client/ethereum.go

```diff
@@ -42,10 +50,24 @@ func NewEthereumClient(network BlockchainNetwork) (*EthereumClient, error) {
 		return nil, err
 	}
 
-	return &EthereumClient{
-		Network: network,
-		Client:  cl,
-	}, nil
+	ec := &EthereumClient{
+		Network:            network,
+		Client:             cl,
+		txQueue:            make(chan common.Hash, 64), // Max buffer of 64 tx
+		queuedTransactions: map[string]EthereumTransactionConfirmer{},
+		mutex:              &sync.Mutex{},
+		queueTransactions:  false,
+		doneChan:           make(chan struct{}),
+	}
+	go ec.newHeadersLoop()
+	return ec, err
+}
+
+// Close tears down the current open Ethereum client
+func (e *EthereumClient) Close() error {
+	e.doneChan <- struct{}{}
+	e.Client.Close()
+	return nil
```

Comment:
Why only return nil? Shouldn't it return the err of `e.Client.Close()`?

Reply:
`Close` has no returns, surprised me as well.